designs/provisioner-priority.md

```diff
@@ -0,0 +1,268 @@
+# Prioritizing Provisioners
+
+## Goals
+
+- Allowing a user to describe a logical ordering to their provisioners
+- Allowing a user to set defaults and/or fallback orderings for their provisioners
+
+## Background
+
+Currently, there is no logical ordering to how the Karpenter scheduling receives and processes Provisioners. Provisioners are processed in the order they are received from the Kubernetes LIST API, which is randomly. Like pod affinities, users want to convey a preferential relationship to some Provisioners over others. Rather than having to manually map a pod to a specific Provisioner through a `nodeSelector` or `affinity` (which could be cumbersome for clusters with large workloads), a user should be able to specify global preferences to pod assignment to provisioners.
```

Comment:
Worth mentioning that we currently recommend that provisioners are mutually exclusive in the docs. We'll need to update this.

Reply:
Sure, I think I mentioned this somewhere down below. Could probably bring this up into the background section.

Reply:
I wonder if that shouldn't still be the recommendation.  Weighting provisioners has some use cases, but in the event that we launch nodes from multiple provisioners in a single scheduling batch with pods that could go to either, we don't control what gets scheduled where.